### DISCUSSION TITLE
jetson nano populate_sdk_ext libconfig-tegra error

### DISCUSSION THREAD

**Discussion by mahdichi** [OE4T/meta-tegra#1495] [Q&A] [answered]
I want build ext sdk for jetson nano in irkstone-l4t-r32.7.x branch but i got stock on this error

ERROR: core-image-minimal-1.0-r0 do_sdk_depends: The file /usr/include/libconfig.h is installed by both libconfig-tegra and libconfig, aborting
ERROR: Logfile of failure stored in: /media/ma/yocto/tegra/build/tmp/work/jetson_nano_devkit_emmc-poky-linux/core-image-minimal/1.0-r0/temp/log.do_sdk_depends.3822732
ERROR: Task (/media/ma/yocto/tegra/src/poky/meta/recipes-core/images/core-image-minimal.bb:do_sdk_depends) failed with exit code '1'

**Comment by madisongh** [comment]
Can you provide more information about your build, like what exactly you're including in your image (e.g., via changes in your `local.conf` file) that might have triggered that error?  When I try an eSDK build with a clean build workspace, I don't see that error.

**Comment by mahdichi** [comment]
this is my local.conf changes

```
#DISTRO_FEATURES = "x11 opengl"
IMAGE_CLASSES += "image_types_tegra"
IMAGE_FSTYPES = "tegraflash"
DL_DIR ?= "/media/ma/yocto/downloads/"
#SSTATE_DIR ?= "${TOPDIR}/sstate-cache"
#TMPDIR = "${TOPDIR}/tmp"
INHERIT += "rm_work"
BB_GENERATE_MIRROR_TARBALLS = "1"
BB_NUMBER_THREADS = "4"
RM_OLD_IMAGE = "1"
#LICENSE_FLAGS_ACCEPTED = "commercial"
IMAGE_INSTALL:append = " gstreamer1.0-plugins-nvvideo4linux2 gstreamer1.0-plugins-nvdrmvideosink gstreamer1.0-plugins-nvv4l2camerasrc gstreamer1.0-plugins-nvvidconv"
IMAGE_INSTALL:append = " gstreamer1.0-plugins-nvarguscamerasrc"
IMAGE_INSTALL:append = " v4l-utils"
IMAGE_INSTALL:append = " l4t-usb-device-mode"
IMAGE_INSTALL:append = " nano"
IMAGE_INSTALL:append = " dropbear"
IMAGE_INSTALL:append = " psplash"
IMAGE_INSTALL:append = " gstreamer1.0-plugins-base" 
IMAGE_INSTALL:append = " gstreamer1.0-plugins-good" 
IMAGE_INSTALL:append = " usbutils util-linux" 
DISTRO_FEATURES:append = " systemd"
VIRTUAL-RUNTIME_init_manager = "systemd"
```

**Reply by madisongh** [reply, marked as answer]
OK, that helps, thanks. You should be able to work around the problem by adding
```
TEGRAFLASH_INITRD_FLASH_IMAGE = ""
```
to your local.conf.  There's an image that gets built for use with the initrd flashing feature that contains a variant of the `libconfig` recipe; looks like I need to tweak that a bit to avoid conflicts for SDK creation.  Turning off that feature will bypass the problem for you.

**Reply by mahdichi** [reply]
thanks, it's OK now.

**Comment by aurelien-enchanted-tools** [comment]
I obtain the same issue with the nanbield branch. But the workaround above does not allow the image to be flashed with ./initrd-flash, because it was not in the "*rootfs.tegraflash.tar.gz" tarball.

**Reply by madisongh** [reply]
Yes, the `libconfig-tegra` recipe still needs fixing to not conflict with the real `libconfig` recipe.

**Reply by aurelien-enchanted-tools** [reply]
I don't understand why and what is the use case which requires in libusbgx-tegra-initrd-flash_git.bb the use of libconfig-tegra ? libconfig-tegra and libconfig are almost identical. If I bbappend libusbgx-tegra-initrd-flash_git.bb with `PACKAGECONFIG[libconfig] = "--with-libconfig=yes,--without-libconfig,libconfig"` it seems to work. I have no issue with `do_populate_sdk_ext` or `./initrd-flash`.

**Reply by madisongh** [reply]
The goal was to make sure we don't have anything more than OE-Core as a layer dependency, so we need to have recipes for things that normally reside in one of the meta-openembedded layers. But I didn't adequately deal with the case where meta-openembedded layers *are* in the build, for those recipes, and that's what I need to fix.

**Reply by madisongh** [reply]
I have updated these recipes in master, scarthgap, scarthgap-l4t-r35.x, kirkstone, and kirkstone-l4t-r32.7.x so they shouldn't conflict with the ones in meta-oe any longer.
